app/feature/feature-profile/src/main/kotlin/com/hedvig/android/feature/profile/settings/NotifyBackendAboutLanguageChangeUseCase.kt

```diff
@@ -3,21 +3,19 @@ package com.hedvig.android.feature.profile.settings
 import com.apollographql.apollo3.ApolloClient
 import com.hedvig.android.apollo.NetworkCacheManager
 import com.hedvig.android.apollo.safeExecute
-import com.hedvig.android.language.LanguageService
 import com.hedvig.android.market.Language
 import giraffe.UpdateLanguageMutation
 
-internal class ChangeLanguageUseCase(
+internal interface NotifyBackendAboutLanguageChangeUseCase {
+  suspend fun invoke(language: Language)
+}
+
+internal class NotifyBackendAboutLanguageChangeUseCaseImpl(
   private val apolloClient: ApolloClient,
-  private val languageService: LanguageService,
   private val cacheManager: NetworkCacheManager,
-) {
-  suspend fun invoke(language: Language) {
-    apolloClient
-      .mutation(
-        UpdateLanguageMutation(language.toString(), language.toLocale()),
-      ).safeExecute()
+) : NotifyBackendAboutLanguageChangeUseCase {
+  override suspend fun invoke(language: Language) {
+    apolloClient.mutation(UpdateLanguageMutation(language.toString(), language.toLocale())).safeExecute()
```

Comment:
We were doing `languageService.setLanguage(language)` here, but this was bringing some confusion as we were as some point calling the languageService function both from this UseCase and calling the real one too.
I figured it's better to just split those up, and call each one separately. This is also important as we do want to make sure we change the language locally first and have that apply instantly, and then do a best effort call to the backend too. Because to be honest I am not even sure why we're doing this mutation in the first place, probably legacy reasons as we should be appending a language header on every invocation anyway, informing the backend for this doesn't make sense to keep for the future.